### PR TITLE
Update OpenMP Hello World

### PR DESCRIPTION
**Pull request by aaronenyeshi**
Add functionality for openmp_helloworld project to build
on Windows, using MSVC's C++ Clang Tools' OpenMP library.

### CODE REVIEW

**Review by ronlieb — APPROVED** [review]

**Review by TorreZuk — COMMENTED** [review]
Okay but this is pulling the MSVC llvm openmp, I didn't install clang support as didn't want to mix two clangs so don't even have a llvm/x64/lib, only llvm/x64/bin (no libomp.dll).   Is the standard MSVC vcomp.dll not compatible?  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\lib\x64\vcomp.lib
Maybe you needed the llvm openmp to launch kernels within the omp threads?   This is fine to install the clang tools but just want to confirm.

File: openmp-helloworld/CMakeLists.txt [hunk 1]
```diff
@@ -3,27 +3,48 @@ project(openmp_helloworld)
...
-# Preparing the executable
+# Preparing the executable.
 add_executable(test_openmp_helloworld ${CPP_SOURCES})
+
+# Link Libraries - HIP Device and OpenMP.
 target_compile_options(test_openmp_helloworld PRIVATE ${OpenMP_CXX_FLAGS})
 target_link_libraries(test_openmp_helloworld PRIVATE hip::device ${OpenMP_CXX_FLAGS})
 
+if(WIN32)
+target_link_libraries(test_openmp_helloworld PUBLIC libomp)
```
Comment (yxsamliu) [inline openmp-helloworld/CMakeLists.txt:49]: can we use target_link_libraries(test_openmp_helloworld PRIVATE OpenMP::OpenMP_CXX)

Comment (aaronenyeshi) [reply openmp-helloworld/CMakeLists.txt:49]: Hi Sam, I've tested that using that also works on Windows. I will update this.

**Comment by aaronenyeshi** [comment]
> Okay but this is pulling the MSVC llvm openmp, I didn't install clang support as didn't want to mix two clangs so don't even have a llvm/x64/lib, only llvm/x64/bin (no libomp.dll). Is the standard MSVC vcomp.dll not compatible? C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\lib\x64\vcomp.lib
> Maybe you needed the llvm openmp to launch kernels within the omp threads? This is fine to install the clang tools but just want to confirm.

It may be more compatible to MSVC's llvm openmp library. But will need time to scope out how to use vcomp instead.

**Comment by TorreZuk** [comment]
As Dan pointed out VS will have an updated OpenMP soon, I don't see this in my version yet: https://devblogs.microsoft.com/cppblog/improved-openmp-support-for-cpp-in-visual-studio/
Should compare against what llvm OpenMP will have...

**Comment by aaronenyeshi** [comment]
> As Dan pointed out VS will have an updated OpenMP soon, I don't see this in my version yet: https://devblogs.microsoft.com/cppblog/improved-openmp-support-for-cpp-in-visual-studio/
> Should compare against what llvm OpenMP will have...

I believe once the default MSVC supports OpenMP with `-openmp:llvm`, we can switch this example to using that instead.

**Comment by aaronenyeshi** [comment]
If this is okay, I will submit it.

**Review by ronlieb — APPROVED** [review]

**Pull request merged**